### PR TITLE
chore: add opencode agent configuration

### PR DESCRIPTION
**Pull request by abaker-9**
## Summary

Adds opencode AI agent configuration for this repository.

- **`opencode.json`** -- main config: loads `AGENTS.md` + copilot instructions, registers plugins, configures the `gopls` LSP, and sets a `permission` block requiring `ask` for `git commit`/`push`/`amend`.
- **`AGENTS.md`** -- primary AI guidance, centered on the git safety & commit-signing policy (ask-first commits, SSH signing + DCO sign-off, ssh-agent key verification for macOS/Linux/Windows).
- **`.opencode/`** -- agents (commit-message, go-reviewer, go-fixer, pr-reviewer, planner, issue-creator), slash commands, and the local `git-signing-guard.ts` plugin that blocks commits when no signing key is loaded. `node_modules`/`package*.json` are excluded via `.opencode/.gitignore`.
- **`.github/copilot-instructions.md`** -- git-safety rule updated to permit AI commits with per-action approval and signing-key verification (pairs with `AGENTS.md`).

## Notes

- The `git-signing-guard` plugin lives in `.opencode/plugins/` (auto-loaded; no config entry required).
- ssh-agent guidance covers macOS (`--apple-use-keychain`), Linux (plain `ssh-add`), and Windows (`Start-Service ssh-agent`).
- All AI docs are ASCII-only per the repo's markdown conventions.

## Test plan

- [ ] opencode loads the config without errors
- [ ] `git-signing-guard` blocks a commit when no key is loaded and allows it once a key is present

### CODE REVIEW

File: .opencode/command/plan.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+description: "scafctl: Create an implementation plan for a scafctl feature. Produces a structured blueprint with architecture decisions, task breakdown, interface design, and testing strategy."
+agent: planner
+---
+Create a structured implementation blueprint for the described feature:
+
+1. **Summary** — What and why
+2. **Architecture decisions** — Layers affected, new types, interface changes
+3. **Task breakdown** — Ordered steps with files, complexity, dependencies
+4. **Interface design** — Define contracts first
+5. **Error handling** — Sentinel errors, wrapping strategy
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply .opencode/command/plan.md]: Fixed in 9f6b64b3984f0e4e215d01e35ab2de7fda9984cb: replaced all em dashes with `--` across the `.opencode` and `.github` AI files to satisfy the ASCII-only markdown rule.

File: .opencode/command/plan.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+6. **Testing strategy** — Unit tests, benchmarks, integration tests (CLI, solution, API)
+7. **Documentation** — Docs, examples, MCP tools, tutorials
+8. **Risks & edge cases** — What could go wrong
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply .opencode/command/plan.md]: Fixed in 9f6b64b3984f0e4e215d01e35ab2de7fda9984cb: replaced all em dashes with `--` across the `.opencode` and `.github` AI files to satisfy the ASCII-only markdown rule.

File: .opencode/agent/planner.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+### 3. Task Breakdown
+Ordered list of implementation steps, each with:
+- What to create/modify
+- Which file(s)
+- Estimated complexity (S/M/L)
+- Dependencies on other tasks
+
+### 4. Interface Design
+Define interfaces FIRST — these are the contracts:
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply .opencode/agent/planner.md]: Fixed in 9f6b64b3984f0e4e215d01e35ab2de7fda9984cb: replaced all em dashes with `--` across the `.opencode` and `.github` AI files to satisfy the ASCII-only markdown rule.

File: .opencode/agent/planner.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+- Security implications?
+- Breaking changes?
+
+## Principles
+
+- **Read-only** — This agent plans but does not modify code
+- **Interface-driven** — Define contracts before implementations
+- **Incremental** — Break work into small, independently testable pieces
+- **Convention-following** — Match existing codebase patterns
+- **Complete** — Include docs, examples, MCP tools, and integration tests in every plan
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply .opencode/agent/planner.md]: Fixed in 9f6b64b3984f0e4e215d01e35ab2de7fda9984cb: replaced all em dashes with `--` across the `.opencode` and `.github` AI files to satisfy the ASCII-only markdown rule.

File: .opencode/agent/issue-creator.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+**Feasibility**: Straightforward or blockers/risks?
+
+**Scope**:
+| Size | Description |
+|------|-------------|
+| **XS** | Trivial — config change, typo fix, single-line edit |
+| **S** | Small — isolated change in 1-2 files, < 1 hour |
+| **M** | Medium — touches multiple files/layers, < 1 day |
+| **L** | Large — cross-cutting change, new interfaces, multi-day |
+| **XL** | Extra large — architectural change, major refactor |
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply .opencode/agent/issue-creator.md]: Fixed in 9f6b64b3984f0e4e215d01e35ab2de7fda9984cb: replaced all em dashes with `--` across the `.opencode` and `.github` AI files to satisfy the ASCII-only markdown rule.

File: .opencode/command/go-build.md [hunk 6]
```diff
@@ -0,0 +1,13 @@
+---
+description: "scafctl: Fix Go build errors, go vet warnings, and linter issues with minimal surgical changes."
+agent: go-fixer
+---
+Diagnose and fix the current Go build errors:
+
+1. Run `go build ./...` to identify compilation errors
+2. Run `go vet ./...` to find vet warnings
+3. Apply minimal, surgical fixes — don't refactor
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply .opencode/command/go-build.md]: Fixed in 9f6b64b3984f0e4e215d01e35ab2de7fda9984cb: replaced all em dashes with `--` across the `.opencode` and `.github` AI files to satisfy the ASCII-only markdown rule.

File: .opencode/command/go-test.md [hunk 7]
```diff
@@ -0,0 +1,13 @@
+---
+description: "scafctl: Run Go tests with race detection, check coverage, and diagnose failures."
+agent: go-fixer
+---
+Run the Go test suite and report results:
+
+1. Run `go test -race -count=1 ./...` — report any failures
+2. Run `go test -coverprofile=coverage.out ./...` — check coverage
+3. Run `go tool cover -func=coverage.out | tail -1` — report total coverage
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply .opencode/command/go-test.md]: Fixed in 9f6b64b3984f0e4e215d01e35ab2de7fda9984cb: replaced all em dashes with `--` across the `.opencode` and `.github` AI files to satisfy the ASCII-only markdown rule.

File: .opencode/agent/issue-creator.md [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+### Phase 3: Assess
+
+Present the user with:
+
+**Feasibility**: Straightforward or blockers/risks?
+
+**Scope**:
+| Size | Description |
+|------|-------------|
+| **XS** | Trivial — config change, typo fix, single-line edit |
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply .opencode/agent/issue-creator.md]: Fixed in 9f6b64b3984f0e4e215d01e35ab2de7fda9984cb: replaced all em dashes with `--` across the `.opencode` and `.github` AI files to satisfy the ASCII-only markdown rule.

File: .opencode/command/artifact-coverage.md [hunk 9]
```diff
@@ -0,0 +1,63 @@
+---
+description: "scafctl: Verify staged changes have updated/added docs, tutorials, examples, solution integration tests, and MCP server tools."
+agent: agent
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply .opencode/command/artifact-coverage.md]: Fixed in 9f6b64b3984f0e4e215d01e35ab2de7fda9984cb: `agent: agent` referenced a non-existent agent. Added a read-only `artifact-auditor` agent (both opencode and Copilot formats) that covers the combined scope, pointed `completeness-check` at it, and removed the duplicate `artifact-coverage` command/prompt.

File: .opencode/plugins/git-signing-guard.ts [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+      const isCommit = /\bgit\b[^&|;]*\bcommit\b/.test(command)
+      if (!isCommit) return
+
+      // Is any identity loaded in the ssh-agent?
+      let hasKey = false
+      try {
+        const res = await $`ssh-add -l`.quiet().nothrow()
+        // exit 0 => at least one key; 1 => agent has no identities;
+        // 2 => cannot contact agent.
+        hasKey = res.exitCode === 0
```
[the remark this answers is not in the document]
Comment (abaker-9) [reply .opencode/plugins/git-signing-guard.ts]: Fixed in 9f6b64b3984f0e4e215d01e35ab2de7fda9984cb: `git-signing-guard` now resolves the repo's configured SSH signing key (`gpg.format=ssh` + `user.signingkey` -> `ssh-keygen -lf` fingerprint) and verifies that specific fingerprint is loaded in the agent, instead of accepting any identity. Falls back to the previous any-key check when SSH signing is not configured.
